### ISSUE TITLE
Filter control select with field formatter does not yield results

### ISSUE THREAD

**Issue by marceloverdijk** [wenzhixin/bootstrap-table#5071]
Having this example: https://live.bootstrap-table.com/code/marceloverdijk/3268

The `country` field is defines as folows:

      <th data-field="country"
          data-filter-control="select"
          data-filter-data="var:countries"
          data-formatter="countryFormatter">
        Country
      </th>

- `data-formatter` is used to display an icon behind the country name
- `data-filter-data` is used as displaying the icon in the select does not work.

 ``` 
  var countries = ['UK', 'US']
  
  var data = [
    { id: 1, name: 'John Doe', country: 'UK' },
    { id: 2, name: 'Jane Doe', country: 'UK' },
    { id: 3, name: 'Alice', country: 'US' },
    { id: 4, name: 'Bob', country: 'US' },
  ]

  function countryFormatter(value, key) {
    return value + ' <i class="fas fa-globe"></i>';
  }
```
  
Now when I e.g. select `US` in the select filter no matches are found.
I guess internally it's comparing `'US' == 'US <i class="fas fa-globe"></i>'` which results in `false`?

I also tried with `data-filter-strict-search="true"`  on the column but with the same result no matches found.

**Comment by UtechtDustin** [comment]
The [filterCustomSearch](https://bootstrap-table.com/docs/extensions/filter-control/#filtercustomsearch) should help you, i guess.

**Comment by marceloverdijk** [comment]
Yes I think [filterCustomSearch](https://bootstrap-table.com/docs/extensions/filter-control/#filtercustomsearch) will help.

But I guess / would assume that it should work without it, right? 
I updated the example a bit: https://live.bootstrap-table.com/code/marceloverdijk/3270
and now for the `name` field I also added a formatter. 
In that case it works as expected but this is an input and not a select.

**Comment by marceloverdijk** [comment]
I was trying with the `filterCustomSearch` and I found out I declared the `data-filter-data` wrongly.
I used:

```
var countries = ['UK', 'US'] 
```
(as I thought it should be an array).
But then the value for `UK` is `1` and for `US` `2`, hence not finding anything...

I changed it to `var countries = {'UK': 'UK', 'US': 'US'}` it does work as expcted:

Updated example: https://live.bootstrap-table.com/code/marceloverdijk/3271

The only question is when defining it as an array like `var countries = ['UK', 'US'] ` shouldn't `bootratrap-table` not use the value as key and value (instead of auto-assigning a sequential number?
